Mathlib/RingTheory/Polynomial/Quotient.lean

```diff
@@ -0,0 +1,247 @@
+/-
+Copyright (c) 2019 Kenny Lau. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kenny Lau, David Kurniadi Angdinata, Devon Tuma, Riccardo Brasca
+
+! This file was ported from Lean 3 source module ring_theory.polynomial.quotient
+! leanprover-community/mathlib commit 61d8b8248633da198afea97ae7a90ee63bdf8c1c
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.Data.Polynomial.Div
+import Mathlib.RingTheory.Polynomial.Basic
+import Mathlib.RingTheory.Ideal.QuotientOperations
+import Mathlib.Tactic.SimpIntro
+
+/-!
+# Quotients of polynomial rings
+-/
+
+
+set_option linter.uppercaseLean3 false
+
+open Polynomial
+
+namespace Polynomial
+
+variable {R : Type _} [CommRing R]
+
+noncomputable def quotientSpanXSubCAlgEquivAux2 (x : R) :
+    (R[X] ⧸ (RingHom.ker (aeval x).toRingHom : Ideal R[X])) ≃ₐ[R] R :=
+let e :=
+  RingHom.quotientKerEquivOfRightInverse (by intro x; exact eval_C :
+    Function.RightInverse (fun a : R => (C a : R[X])) (@aeval R R _ _ _ x))
+{ e with commutes' := fun r => e.apply_symm_apply r }
+
+noncomputable def quotientSpanXSubCAlgEquivAux1 (x : R) :
+  (R[X] ⧸ Ideal.span {X - C x}) ≃ₐ[R] (R[X] ⧸ (RingHom.ker (aeval x).toRingHom : Ideal R[X])) :=
+@Ideal.quotientEquivAlgOfEq R R[X] _ _ _ _ _ (ker_evalRingHom x).symm
+
+noncomputable def quotientSpanXSubCAlgEquiv (x : R) :
```

Comment:
As mentioned above, this has been splited into two sub-constructions which may or may not be ideal